### PR TITLE
Patch to fix esmpy import for esmpy >= 8.4.0

### PR DESCRIPTION
**Pull request by xylar**
Add a test to make sure `regrid2.ESMFREgrid` is available.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Equivalent to #87 but for the `main` branch.

**Comment by xylar** [comment]
@conda-forge-admin, please restart CI

File: recipe/meta.yaml [hunk 1]
```diff
@@ -69,6 +70,7 @@ test:
     - python -c 'import regrid2'
     - python -c 'import regrid2._regrid'
     - python -c 'import regrid2._scrip'
+    - python -c 'from regrid2 import ESMFRegrid'
```
Comment (xylar) [inline recipe/meta.yaml:73]: @chengzhuzhang, hopefully this check will prevent issues like https://github.com/E3SM-Project/e3sm_diags/issues/711 from happening in the future.

Comment (chengzhuzhang) [reply recipe/meta.yaml:73]: It looks like regridding behavior gets back to normal with this fix!

**Pull request merged**